linesearch.go

```diff
@@ -63,46 +60,34 @@ func (ls *LinesearchMethod) Iterate(loc *Location) (Operation, error) {
 		// Init was not called. What to do? What about ls's internal state?
```

Comment:
I still don't understand this comment. What does "and init was not called" mean? Do you mean "or init was not called"? It seems like we can get in this state in some error cases or if init was not called. I think the right thing is to document that if an error was returned the LinesearchMethod must be re-initialized. For example, if the linesearch fails the Method may try again in a different direction. This should be fine, just Init needs to be called again (with a valid location).